x-pack/plugins/security_solution/server/plugin.ts

```diff
@@ -225,27 +225,33 @@ export class Plugin implements IPlugin<PluginSetup, PluginStart, SetupPlugins, S
         }
 
         await ruleDataService.createOrUpdateComponentTemplate({
-          name: componentTemplateName,
-          body: {
-            template: {
-              settings: {
-                number_of_shards: 1,
+          template: {
+            name: componentTemplateName,
+            body: {
+              template: {
+                settings: {
+                  number_of_shards: 1,
+                },
+                mappings: {}, // TODO: Add mappings here via `mappingFromFieldMap()`
               },
-              mappings: {}, // TODO: Add mappings here via `mappingFromFieldMap()`
             },
           },
+          templateVersion: 1,
         });
 
         await ruleDataService.createOrUpdateIndexTemplate({
-          name: ruleDataService.getFullAssetName('security-solution-index-template'),
-          body: {
-            index_patterns: [ruleDataService.getFullAssetName('security-solution*')],
-            composed_of: [
-              ruleDataService.getFullAssetName(TECHNICAL_COMPONENT_TEMPLATE_NAME),
-              ruleDataService.getFullAssetName(ECS_COMPONENT_TEMPLATE_NAME),
-              componentTemplateName,
-            ],
+          template: {
+            name: ruleDataService.getFullAssetName('security-solution-index-template'),
+            body: {
+              index_patterns: [ruleDataService.getFullAssetName('security-solution*')],
+              composed_of: [
+                ruleDataService.getFullAssetName(TECHNICAL_COMPONENT_TEMPLATE_NAME),
+                ruleDataService.getFullAssetName(ECS_COMPONENT_TEMPLATE_NAME),
```

Comment:
Probably an offtopic, but not sure where to better discuss this.

What's the intent behind this separation to "technical" template and "ecs" template? Why we need to have 2 templates? Imho everything in them is just a bunch of common mappings, and it could be specified in a single template.

I also checked the templates, and it's weird to me:

```ts
export const technicalComponentTemplate: ClusterPutComponentTemplateBody = {
  template: {
    settings: {
      number_of_shards: 1,
    },
    mappings: mappingFromFieldMap(technicalRuleFieldMap),
  },
};

export const technicalRuleFieldMap = {
  ...pickWithPatterns(
    ecsFieldMap,
    TIMESTAMP,
    EVENT_KIND,
    EVENT_ACTION,
    RULE_UUID,
    RULE_ID,
    RULE_NAME,
    RULE_CATEGORY,
    TAGS
  ),
  [PRODUCER]: { type: 'keyword' },
  [ALERT_UUID]: { type: 'keyword' },
  [ALERT_ID]: { type: 'keyword' },
  [ALERT_START]: { type: 'date' },
  [ALERT_END]: { type: 'date' },
  [ALERT_DURATION]: { type: 'long' },
  [ALERT_SEVERITY_LEVEL]: { type: 'keyword' },
  [ALERT_SEVERITY_VALUE]: { type: 'long' },
  [ALERT_STATUS]: { type: 'keyword' },
  [ALERT_EVALUATION_THRESHOLD]: { type: 'scaled_float', scaling_factor: 100 },
  [ALERT_EVALUATION_VALUE]: { type: 'scaled_float', scaling_factor: 100 },
} as const;
```

```ts
export const ecsComponentTemplate: ClusterPutComponentTemplateBody = {
  template: {
    settings: {
      number_of_shards: 1,
    },
    mappings: merge(
      {},
      mappingFromFieldMap(ecsFieldMap),
      mappingFromFieldMap(technicalRuleFieldMap)
    ),
  },
};
```

- So ecs template contains ALL the ecs fields + technical fields.
- Technical template contains technical fields.
- Technical fields are a subset of ecs fields + some extensions to ecs (our custom fields).

And the resulting index template references all of this, thus essentially includes ALL standard ECS fields + our extensions. Is this our plan to include the whole ECS?

Also, wouldn't it be better to have 2 separate field maps (one for a common subset of ECS fields and another for extensions to ECS), and then merge them into a single common `.alerts-mappings` component template?

This is what I basically did in the draft (https://github.com/elastic/kibana/pull/101453) and described in https://github.com/elastic/kibana/issues/101016#issuecomment-857593741. Would be happy to get any comments on this.

Reply:
Yeah it seems odd that the `ecsComponentTemplate` contains the `technicalRuleFieldMap` - maybe that's an oversight? I think the overall idea is that the `technicalComponentTemplate` contains the fields that are common across all rules in the RAC initiative, so we may want to enforce that that template is used in all index templates created by the RuleDataPluginService (maybe by automatically dropping it into the `composed_of` field). The `ecsComponentTemplate` with all ECS fields will be useful for security solution templates, since we include the entire ECS mapping so that when we copy documents into the `.alerts` index the ECS fields in the original documents will be mapped in the alert documents as well. We'll most likely have security specific fields to map in the `.alerts-security` indices also, and we could put those in a third component template (first 2 are technical and ECS components).

Re: merging the field maps, I would like to leave the merging of component templates to Elasticsearch, similar to how it's described in https://github.com/elastic/kibana/issues/101016#issuecomment-857593741 with a variety of separate component templates, and avoid constructs like 
```
mappings: merge(
      {},
      mappingFromFieldMap(ecsFieldMap),
      mappingFromFieldMap(technicalRuleFieldMap)
    ),
```
where we explicitly merge mappings into single component. I like the pattern we use to build the security solution index template right now where we compose the full index template using the base technical template, the ECS template, and our (currently empty) security solution-specific component template:
```
{
  name: ruleDataService.getFullAssetName('security-solution-index-template'),
  body: {
    index_patterns: [ruleDataService.getFullAssetName('security-solution*')],
    composed_of: [
      ruleDataService.getFullAssetName(TECHNICAL_COMPONENT_TEMPLATE_NAME),
      ruleDataService.getFullAssetName(ECS_COMPONENT_TEMPLATE_NAME),
      componentTemplateName,
    ],
  },
}
```
Since Observability indices don't want the full ECS mapping, we need to keep the base `.alerts` component template as minimal as possible and allow solutions to add various components as needed.

Reply:
Got it. Yeah, maybe there's an oversight and some confusion. I see your points. Let's maybe discuss this separately and address in a follow-up PR (I could take over it).